x-pack/plugins/cases/public/components/connectors/servicenow/servicenow_sir_case_fields.test.tsx

```diff
@@ -172,15 +180,31 @@ describe('ServiceNowSIR Fields', () => {
     expect(screen.queryByTestId('deprecated-connector-warning-callout')).not.toBeInTheDocument();
   });
 
+  test('it should hide subcategory if selecting a category without subcategories', async () => {
+    // Failed Login doesn't have defined subcategories
+    const customFields = {
+      ...fields,
+      category: 'Failed Login',
+      subcategory: '',
+    };
+    const wrapper = mount(
+      <Fields fields={customFields} onChange={onChange} connector={connector} />
+    );
+
+    expect(wrapper.find('[data-test-subj="subcategorySelect"]').exists()).toBeFalsy();
+  });
+
   describe('onChange calls', () => {
-    const wrapper = mount(<Fields fields={fields} onChange={onChange} connector={connector} />);
+    let wrapper: ReactWrapper;
```

Comment:
note: moving some of this code that was inside the describe function into a `beforeEach` to prevent tests cross-contamination.